### PR TITLE
Tests/minor gaps

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

PR addressing various small gaps in test coverage

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: tests/test_protocols/test_dialogue/test_base.py [hunk 1]
```diff
@@ -1834,7 +1830,7 @@ def test_cleanup(self):
         assert dialogues_storage._incomplete_to_complete_dialogue_labels
         assert dialogues_storage._terminal_state_dialogues_labels
 
-        self.dialogues._dialogues_storage.cleanup()
+        self.dialogues.cleanup()
```
Comment (Karrenbelt) [inline tests/test_protocols/test_dialogue/test_base.py:1833]: effectively amounts to the same but increases test coverage

File: tests/test_runtime.py [hunk 2]
```diff
@@ -128,6 +131,31 @@ async def error(*args, **kwargs):
 
         assert self.runtime.state == RuntimeStates.error, self.runtime.state
 
+    def test_cancelled_during_start_agent_loop(self, caplog):
+        """Test asyncio.CancelledError during _start_agent_loop."""
+
+        # we check for this log because async_utils.Runnable._thread_target
+        # catches all BaseExceptions and logs them instead of raising
```
Comment (Karrenbelt) [inline tests/test_runtime.py:138]: this one was a bit nasty to write, got some useful pair programming assistance from @0xArdi and left a comment for the reader to explain why `pytest.raises` cannot be used to catch the `asycio.CancelledError`

File: tests/test_runtime.py [hunk 3]
```diff
@@ -128,6 +131,31 @@ async def error(*args, **kwargs):
...
+                wait_for_condition(
+                    lambda: f"Exception raised in {self.runtime}" in caplog.text,
+                    timeout=2,
+                    error_msg="Failed to raise asyncio.CancelledError during _start_agent_loop.",
+                )
+
+        # this is needed else will be raised in teardown
+        with contextlib.suppress(asyncio.CancelledError):
+            self.runtime.stop()
+            self.runtime.wait_completed(sync=True)
```
Comment (Karrenbelt) [inline tests/test_runtime.py:157]: this was a bonus

**Review by Adamantios — APPROVED** [review]

File: tests/test_protocols/test_dialogue/test_base.py [hunk 4]
```diff
@@ -1766,10 +1766,6 @@ def test_dump_restore(self):
             dialogues_storage_restored._dialogue_by_address
         )
 
-        assert len(dialogues_storage._dialogue_by_address) == len(
```
Comment (DavidMinarsch) [inline tests/test_protocols/test_dialogue/test_base.py:1769]: why removed?

Comment (Karrenbelt) [reply tests/test_protocols/test_dialogue/test_base.py:1769]: https://github.com/valory-xyz/open-aea/blob/68c2b47f039c7c20aa81987998814e328f4619e5/tests/test_protocols/test_dialogue/test_base.py#L1765-L1771

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**